applications/CoSimulationApplication/tests/dummy_solver/cpp/dummy_solver.cpp

```diff
@@ -0,0 +1,336 @@
+}
+
+void ImportData2(CoSim::CoSimIO& rCoSimIO, DataFieldType& rDataField, const std::string& rIdentifier)
+{
+    CoSim::DataContainers::Data data = {rDataField};
+    rCoSimIO.Import(data, rIdentifier);
+}
+
+void ExportData2(CoSim::CoSimIO& rCoSimIO, DataFieldType& rDataField, const std::string& rIdentifier)
+{
+    CoSim::DataContainers::Data data = {rDataField};
+    rCoSimIO.Export(data, rIdentifier);
+}
+
+void RunSolutionLoop(MeshType& rMesh, DataFieldType& rDataField)
+{
+    double current_time = 0.0;
+    const double end_time = 0.49;
+    while (current_time<end_time) {
+        current_time = AdvanceInTime(current_time);
+        InitializeSolutionStep();
+        Predict();
+        SolveSolutionStep();
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+}
+
+void RunSolutionLoopWithWeakCoupling(MeshType& rMesh, DataFieldType& rDataField)
+{
+    // Note the following only works with one coupling inteface, requires more effort to make it work with multiple coupling interfaces.
+
+    CoSim::CoSimIO co_sim_io("dummy_solver_cpp", "dummy_solver_io_settings");
+
+    co_sim_io.Connect();
+    ExportMesh2(co_sim_io, rMesh, "interface"); // send the coupling-interface mesh to be used for e.g. mapping
+
+    double current_time = 0.0;
+    const double end_time = 0.49;
+    while (current_time<end_time) {
+        current_time = AdvanceInTime(current_time);
+        InitializeSolutionStep();
+        Predict();
+
+        ImportData2(co_sim_io, rDataField, "interface_temp");
+        SolveSolutionStep();
+        ExportData2(co_sim_io, rDataField, "interface_pressure");
+
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+
+    co_sim_io.Disconnect();
+}
+
+void RunSolutionLoopWithStrongCoupling(MeshType& rMesh, DataFieldType& rDataField)
+{
+    // Note the following only works with one coupling inteface, requires more effort to make it work with multiple coupling interfaces.
+
+    CoSim::CoSimIO co_sim_io("dummy_solver_cpp", "dummy_solver_io_settings");
+
+    co_sim_io.Connect();
+    ExportMesh2(co_sim_io, rMesh, "interface"); // send the coupling-interface mesh to be used for e.g. mapping
+
+    double current_time = 0.0;
+    const double end_time = 0.49;
+    while (current_time<end_time) {
+        current_time = AdvanceInTime(current_time);
+        InitializeSolutionStep();
+        Predict();
+        while(true) {
+            ImportData2(co_sim_io, rDataField, "interface_temp");
+            SolveSolutionStep();
+            ExportData2(co_sim_io, rDataField, "interface_pressure");
+            if (co_sim_io.IsConverged()) {
+                break;
+            }
+        }
+
+        FinalizeSolutionStep();
+        OutputSolutionStep();
+        std::cout << std::endl;
+    }
+
+    co_sim_io.Disconnect();
+}
+
+void RunSolutionCoSimulationOrchestrated(MeshType& rMesh, DataFieldType& rDataField)
```

Comment:
this function is the full integration of CoSim into an external solver

(ignore the `new`/`delete`, this is a temp hack)